### PR TITLE
Switch to pytest

### PR DESCRIPTION
**Pull request by amureki**
- Renamed `test` -> `tests`
- Removed `runtests.py` to simplify test runs via `pytest` (django app is being set in `conftest` now)
- Removed `django-test-without-migrations` in favor of `pytest-django`'s `--nomigrations`
- Updated `tox.ini` with new dependencies

Please, note, that this PR does not contain nice `pytest` tweaks yet, but just reuses current tests as is. I did not want to overcomplicate PR, it is already too big to have a quick look.

## How to review/test

- Compare [previous travis run](https://travis-ci.org/berinhard/model_mommy/builds/498384081) with the current one, see that amount of tests and overall output is the same
- (???) see that I removed `create database model_mommy;` and `create database model_mommy_test;`, so it uses only `postgres` database - maybe you could see downsides of this (and reasoning behind the original logic)
- locally install dev dependencies and do `pytest`

### CODE REVIEW

**Comment by amureki** [comment]
@anapaulagomes @berinhard what do you think about this change? I strongly believe, pytest is a nice alternative to a default test runner with nice tweaks.

File: .travis.yml [hunk 1]
```diff
@@ -21,11 +21,9 @@ addons:
...
   - postgresql
 before_script:
-  - psql -U postgres -c "create extension postgis"
-  - psql -U postgres -c "create extension hstore"
-  - psql -U postgres template1 -c "create extension citext"
-  - psql -c 'create database model_mommy;' -U postgres
-  - psql -c 'create database model_mommy_test;' -U postgres
+  - psql template1 -c "create extension hstore;"
+  - psql template1 -c "create extension postgis;"
+  - psql template1 -c "create extension citext"
```
Comment (anapaulagomes) [inline .travis.yml]: Shall we add a `;` here only for consistency's sake? 😅

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import os
+
+import django
+from django.conf import settings
+
+
+def pytest_configure():
+    test_db = os.environ.get('TEST_DB', 'sqlite')
```
Comment (anapaulagomes) [inline tests/conftest.py]: In my opinion makes sense having `sqlite` as the default database but would be nice to add this error `E   django.db.utils.OperationalError: unrecognized token: ":"` in our [contributing guide](https://model-mommy.readthedocs.io/en/latest/#contributing) at some point.

File: tox.ini [hunk 3]
```diff
@@ -6,21 +6,26 @@ envlist =
...
-    django-test-without-migrations
     django111: Django>=1.11,<1.12
     django20: Django==2.0
     django21: Django>=2.1
-    postgresql: psycopg2
+    postgresql: psycopg2-binary
 commands =
-    postgresql: {toxinidir}/runtests.py --postgis
-    sqlite: {toxinidir}/runtests.py --use-tz
+    pytest
```
Comment (anapaulagomes) [inline tox.ini]: ❤️

File: tests/conftest.py (hunk 2 above)
Comment (amureki) [reply tests/conftest.py]: I just removed psycopg2 from dev dependency (it is anyways going to be installed via tox). So it should not be a problem for new users.

We might rethink the whole concept of dev_requirements.txt and if we still need it.

**Pull request merged**